packages/driver/cypress/e2e/commands/actions/type_events.cy.js

```diff
@@ -594,29 +594,48 @@ describe('src/cy/commands/actions/type - #type events', () => {
       ]
 
       targets.forEach((target) => {
+        // NOTE: newer versions of firefox do NOT trigger a click event on buttons when typed into
```

Comment:
I tested this out in Firefox (outside of Cypress) and buttons still trigger click events when typing anything that triggers the button's action (e.g. `space` or `enter`). If Cypress is not issuing the click in the same circumstance, it seems like a bug in Cypress.

Reply:
what version of Firefox did you test on?

Reply:
The latest, 117

Reply:
Is it documented somewhere that newer versions don't issue a click event for typing into buttons?

Reply:
It isn't likely because its so hyper-specific, which makes we wonder if you are right. I had some issues reproducing but I am on an older version, so I am going to try updating again and see if I can reproduce it.

Reply:
Yeah, I think this would be a pretty big breaking change for Firefox to make since I imagine a good number of websites rely on hitting enter/space on a button triggering a click event.

Reply:
I tried again to reproduce on 117 and still can't get it typing into the button to trigger a click event. I haven't confirmed that this works on an older version on my machine but I wonder if there is something wrong with my reproduction?

Reply:
@chrisbreiding has been investigating and looks like click was removed on enter in firefox 106. I will work on adding a synthetic event in to fire the click ourselves if greater than or equal to firefox 106 and the keypress is enter